ai/speech_optimization_enforcement.py

```diff
@@ -37,17 +37,10 @@ async def enforce_speech_optimization(message, message_copy):
     # Strip message attachments of optimized drone.
     message_copy.attachments = []
 
-    # Check for status message.
-    status_message = status_code_regex.match(message_copy.content)
+    status_type, _, _ = get_status_type(message_copy.content)
 
-    if status_message is None:
-        # Optimized drone has posted a non-status message. Delete.
-        LOGGER.info("Optimized drone posted non-status code message. Deleting.")
-        await message.delete()
-        return True
-
-    if get_status_type(status_message) not in (StatusType.PLAIN, StatusType.ADDRESS_BY_ID_PLAIN):
-        LOGGER.info("Optimized drone has posted an informative status message. Deleting.")
+    if status_type not in (StatusType.PLAIN, StatusType.ADDRESS_BY_ID_PLAIN) or status_type == StatusType.NONE:
+        LOGGER.info("Optimized drone has posted an informative or otherwise inappropriate status message. Deleting.")
         # Optimized drone posted an informative status message. Delete.
```

Comment:
This is just it nitpicking, but this comment should probably be reworded and moved up a couple of lines.